### PR TITLE
Correctly handle null S3 object version ID in c7n_salactus

### PR DESCRIPTION
**Pull request by ewbankkit**
Fixes https://github.com/capitalone/cloud-custodian/issues/2003.

* Correctly set 'IsLatest' to false for old object versions
* Correctly handle null object version ID
* Add unit test

### CODE REVIEW

**Comment by ewbankkit** [comment]
The unit test verifies that if there are unencrypted objects in a bucket before versioning is enabled on that bucket then the unencrypted versions are encrypted via the `EncryptExtantKeys` code.

**Comment by alfredgamulo** [comment]
lgtm

File: tests/test_s3.py [hunk 1]
```diff
@@ -1906,6 +1906,46 @@ def test_encrypt_versioned_bucket(self):
             'ServerSideEncryption' in client.head_object(
                 Bucket=bname, Key='home.txt'))
 
+    def test_encrypt_versioned_bucket_with_existing_keys(self):
```
Comment (kapilt) [inline tests/test_s3.py]: can you add the functional decorator, the test looks context free

**Comment by ewbankkit** [comment]
`@functional` added to unit test.

**Pull request merged**